src/app/app.config.ts

```diff
@@ -1,10 +1,11 @@
 import { ApplicationConfig, provideZoneChangeDetection } from '@angular/core';
-import { provideRouter } from '@angular/router';
-
-import { routes } from './app.routes';
+import { provideRouter, withComponentInputBinding } from '@angular/router';
 import { provideHttpClient } from '@angular/common/http';
 import { AppService } from './app.service';
 
+import { AppRoutes } from './app.routes';
+
+
 export const appConfig: ApplicationConfig = {
-  providers: [provideZoneChangeDetection({ eventCoalescing: true }), provideRouter(routes), provideHttpClient(), AppService]
+  providers: [provideZoneChangeDetection({ eventCoalescing: true }), provideRouter(AppRoutes, withComponentInputBinding()), provideHttpClient(), AppService]
```

Comment:
```suggestion
  providers: [provideZoneChangeDetection({ eventCoalescing: true }), provideRouter(AppRoutes), provideHttpClient(), AppService]
```
as you're using `paramMap.get` in the component, you can take this bit out & it'll carry on working (interesting though as I actually didn't know this was a thing!)